ocaml/runtime/fiber.c

```diff
@@ -175,23 +187,19 @@ Caml_inline struct stack_info* alloc_for_stack (mlsize_t wosize)
   // -------------------- <- page-aligned
   // ... (for alignment)
   // struct stack_info
-  // -------------------- <- block, possibly unaligned
-
-  struct stack_info* block;
-  block = mmap(NULL, len, PROT_NONE, MAP_PRIVATE | MAP_ANONYMOUS | MAP_STACK,
-               -1, 0);
-  if (block == MAP_FAILED) {
-    return NULL;
-  }
-  if (madvise (block, len, MADV_NOHUGEPAGE)) {
-    munmap(block, len);
-    return NULL;
-  }
-  if (mprotect(block, len, PROT_READ | PROT_WRITE)) {
-    munmap(block, len);
+  // -------------------- <- [block], possibly unaligned
+  // 2Mb (= extra_size)
+  // -------------------- <- [stack], returned from [mmap]
+  char* stack;
+  stack = mmap(NULL, len + extra_size,
+    PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS,
+    -1, 0);
+  if (stack == MAP_FAILED) {
     return NULL;
   }
 
+  struct stack_info* block = (struct stack_info*) (stack + extra_size);
```

Comment:
It looks like you're throwing away the original `stack` value, which means the `munmap` in `caml_free_stack` isn't going to do the right thing.